prometheus-pushgateway/prometheus-pushgateway.spec

```diff
@@ -0,0 +1,209 @@
+
+Requires(pre):    shadow-utils
+Requires(post):   systemd
+Requires(preun):  systemd
+Requires(postun): systemd
+%else
+Requires:         initscripts kaosv >= 2.15
+
+Requires(pre):    shadow-utils
+Requires(post):   %{__chkconfig}
+Requires(preun):  %{__chkconfig} %{__service}
+Requires(postun): %{__service}
+%endif
+
+Provides:         %{name} = %{version}-%{release}
+
+################################################################################
+
+%description
+The Prometheus Pushgateway exists to allow ephemeral and batch jobs to expose
+their metrics to Prometheus. Since these kinds of jobs may not exist long
+enough to be scraped, they can instead push their metrics to a Pushgateway.
+The Pushgateway then exposes these metrics to Prometheus.
+
+The Pushgateway is explicitly not an aggregator or distributed counter but
+rather a metrics cache. It does not have statsd-like semantics. The metrics
+pushed are exactly the same as you would present for scraping in a permanently
+running program.
+
+For machine-level metrics, the textfile collector of the Node exporter is
+usually more appropriate. The Pushgateway is intended for service-level metrics.
+
+The Pushgateway is not an event store. While you can use Prometheus as a data
+source for Grafana annotations, tracking something like release events has to
+happen with some event-logging framework.
+
+################################################################################
+
+%prep
+%setup -q -n %{pkg_name}-%{version}
+
+mkdir -p .src/github.com/prometheus/%{pkg_name}
+mv * .src/github.com/prometheus/%{pkg_name}/
+mv .promu.yml .src/github.com/prometheus/%{pkg_name}/
+mv .src src
+
+%build
+export GOPATH=$(pwd)
+
+pushd src/github.com/prometheus/%{pkg_name}
+    %{__make} %{?_smp_mflags} build
+popd
+
+%install
+rm -rf %{buildroot}
+
+install -dm 0755 %{buildroot}%{_bindir}
+install -dm 0755 %{buildroot}%{_sysconfdir}/%{name}
+install -dm 0755 %{buildroot}%{_sysconfdir}/logrotate.d
+install -dm 0755 %{buildroot}%{_sysconfdir}/sysconfig
+install -dm 0755 %{buildroot}%{_logdir}/%{name}
+
+%if 0%{?rhel} >= 7
+    install -dm 0755 %{buildroot}%{_unitdir}
+%else
+    install -dm 0755 %{buildroot}%{_initrddir}
+%endif
+
+install -pm 0755 src/github.com/prometheus/%{pkg_name}/%{pkg_name} %{buildroot}%{_bindir}/%{name}
+install -pm 0644 %{SOURCE3} %{buildroot}%{_sysconfdir}/sysconfig/%{name}
+install -pm 0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
+
+%if 0%{?rhel} >= 7
+    install -pm 0644 %{SOURCE1} %{buildroot}%{_unitdir}/%{name}.service
+%else
+    install -pm 0755 %{SOURCE2} %{buildroot}%{_initrddir}/%{name}
+%endif
+
+%clean
+rm -rf %{buildroot}
+
+################################################################################
+
+%pre
+%{__getent} group %{service_group} >/dev/null || %{__groupadd} -r %{service_group}
+%{__getent} passwd %{service_user} >/dev/null || %{__useradd} -r \
+    -g %{service_group} -d %{_sharedstatedir}/%{name} \
```

Comment:
You have the same user in `prometheus` package, but in this package, you try to create the user with a different home directory.

Reply:
Oops, my bad. It seems that `/var/lib/prometheus` will never exist whether case user install only Pushgateway. So I decided to change the username, group and home directory. (b845430e2f66ba0f0ff1ef2e9cae99a577b80c4b)